src/Microsoft.Health.Fhir.Core/Features/Search/SearchValues/ReferenceSearchValueParser.cs

```diff
@@ -40,7 +40,7 @@ public ReferenceSearchValueParser(IFhirRequestContextAccessor fhirRequestContext
         public ReferenceSearchValue Parse(string s)
         {
             EnsureArg.IsNotNullOrWhiteSpace(s, nameof(s));
-
+            s = s.UnescapeSearchParameterValue();
```

Comment:
I'm a little surprised this didn't break a unit test also :) i.e. https://github.com/microsoft/fhir-server/blob/master/src/Microsoft.Health.Fhir.Shared.Core.UnitTests/Features/Search/SearchValues/ReferenceSearchValueParserTests.cs

Reply:
I've look and it seems like there is regex in reference search parameter which cuts values and doesn't let any special characters like ',' or `|` or `$`
I'm a bit scary to modify that regex, so I'm gonna be defensive and remove my changes in reference search param

Reply:
I think that sounds ok. The validation for an id (https://www.hl7.org/fhir/datatypes.html) looks like it can't include these